### PR TITLE
[base] 基礎建設微調 Updates `shallow` for testing async functions

### PR DESCRIPTION
**Pull request by rwu823**
@chuanxd @CApopsicle 
- 調整 `webpack` 更新
- 新增 `shallow` 測試 async code
- 把 build status badges 設成 `dev` only

### CODE REVIEW

**Review by CApopsicle — APPROVED** [review]
LGTM

File: src/components/select/__test__/select.spec.js [hunk 1]
```diff
@@ -1,34 +1,64 @@
...
-
 it('Test <Select /> props `name` and `age`', ()=> {
-  const vm = render((h)=> {
-    return (
-      <Select name="Jimy" age={300} />
-    )
+  const vm = shallow({
+    render(h) {
+      return (
+        <Select name="Jimy" age={20} />
```
Comment (chuanxd) [inline src/components/select/__test__/select.spec.js:35]: not good - -

Comment (rwu823) [reply src/components/select/__test__/select.spec.js:35]: `age={50}` ?

File: src/components/select/__test__/select.spec.js [hunk 2]
```diff
@@ -1,34 +1,64 @@
...
+    render(h) {
+      return (
+        <Select />
+      )
+    }
   })
 
   vm.handleClick()
+  await new Promise((done)=> {
+    vm.$nextTick(done)
```
Comment (chuanxd) [inline src/components/select/__test__/select.spec.js:26]: `In order to wait until Vue.js has finished updating the DOM after a data change.`
http://vuejs.org/guide/reactivity.html#Async-Update-Queue

**Review by chuanxd — APPROVED** [review]

**Comment by chuanxd** [comment]
LGTM

**Pull request merged**